woocommerce/woocommerce-ios
# Update `ProductUpdater` with Copy pattern

Fixes #2433 

## Changes

Thanks to @shiki's awesome work on the Copy pattern & codegen, we can now remove the verbose `ProductUpdater`!

- Updated `Product` to conform to `GeneratedCopiable`, the diffs in `Models+Copiable.generated` were autogenerated via `rake generate`
- Replaced `ProductUpdater` usage in `ProductFormViewModel` with `copy` on the affected properties
- Added unit tests for `ProductFormViewModel`'s `update*` functions from the previous `Product+UpdaterTestCases`
- Removed `ProductUpdater` and its unit tests in `Product+UpdaterTestCases`

## Testing

No app changes should be introduced, please feel free to sanity check product editing:

- Go to the Products tab
- Tap on a simple Product
- Edit a few different fields for the product
- Tap "Update" --> the product should be updated remotely

Update release notes:
